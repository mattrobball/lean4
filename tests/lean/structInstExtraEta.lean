/-! This test assures that that `toA` field is not eta expanded using the `with` pattern as it was
  before #2478 -/

structure A where

structure B extends A

structure C extends B

structure D extends B

def a : A := sorry

def b : B := { a with }
#print b

def c : C := { a with }
#print c

def d : D := { c with }
#print d
