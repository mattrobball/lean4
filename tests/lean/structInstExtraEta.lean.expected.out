structInstExtraEta.lean:12:4-12:5: warning: declaration uses 'sorry'
def b : B :=
let src := a;
{ toA := src }
def c : C :=
let src := a;
{ toB := { toA := src } }
def d : D :=
let src := c;
{ toB := src.toB }
